pkg/render/logstorage.go

```diff
@@ -1419,3 +1395,60 @@ func (es elasticsearchComponent) kibanaPodSecurityPolicy() *policyv1beta1.PodSec
 	psp.GetObjectMeta().SetName("tigera-kibana")
 	return psp
 }
+
+// If user has set ResourceRequirements, replace individual field's default value with user's value.
+// If user provided both Limits and Requests, use them.
+// As ResourceRequirements.Limits must be >= ResourceRequirements.Requests,
+// if user provided just Limits, and Limits is <= default Requests, set Requests value as user's Limits value,
+// if user provided just Requests, and Requests is >= default Limits, set Limits value as user's Requests value.
+func overrideResourceRequirements(defaultReq *corev1.ResourceRequirements, userOverrides corev1.ResourceRequirements) {
```

Comment:
Instead of using some pointer magic, could just return a resource requiment object? There's no real benefit over returning an object, and these defaults are no long defaults after this.

If I was to factor out this logic into a function, I would have a declaration like this:
`func calculateResourceRequirements(defaultReq *corev1.ResourceRequirements, userProvided *corev1.ResourceRequirements) corev1.ResourceRequirements`.

Could you also clean up the function comment, it seems like maybe this should be bullet pointed, with a starting comment like "calculateResourceRequirements combines the default resources with the user specified resources as with the following rules:".